src/vocabularyChecker/detectWrongSyntax.ts

```diff
@@ -0,0 +1,91 @@
+import json from "./AllowedVocab.json";
+import Issue from "./Issue";
+/**
+ * Regular expression that looks for any and all words. A word is defined as some sequence of
+ * characters (\w*S) betweeen wordbarriers (\b)
+ */
+const anyWord = new RegExp(/\b(\w*\S)\b/g);
+/**
+ * Predefined dictionary containing all allowed Words
+ */
+const allowedWords = json;
+interface Position { fromIndex: number; toIndex: number; }
+
+/**
+ * @param text
+ * @param invalidWords
+ * Returns a map of invalid words (consisted in @param invalidWords) and their corresponding
+ * positions (starting position and end position) inside of a string.
+ */
+
+export function createMapOfInvalidWords(text: string, invalidWords: string[]): Map<string, Position[]> {
+	const positions = new Map<string, Position[]>();
+	for (const word of invalidWords) {
+		positions.set(word,  text.indexOf(word), (text.indexOf(word) + word.length - 1);
+	}
+	return positions;
+}
+
+/**
+ * @param text
+ * @param invalidWords
+ * Returns message that tells the user which words are not allowed and on which
+ * position those are within @param text.
+ */
+export function createErrorMessages(text: string, invalidWords: string[]): string|undefined {
+	const wrongWords = logInvalidWords(createMapOfInvalidWords(text, invalidWords));
+	return wrongWords;
+}
+
+/**
+ * @param text
+ * Just a function to bundle the functionalities of @function collectAllInvalidWords and
+ * @function removeDuplicates.
+ * @returns a Stringarray containing exactly one copy of each wrong word in @param text.
+ */
+export function getInvalidWords(text: string): string[] {
+	return removeDuplicates(collectAllInvalidWords(text));
+}
+
+/**
+ * @param text
+ * @returns a Stringarray containing every occurrence of each wrong word in @param text.
+ */
+export function collectAllInvalidWords(text: string): string[] {
+	const words = text.match(anyWord) || [];
```

Comment:
`[...text.matchAll(anyword)]` will give you the index of every match as well. You can then return the invalid words and their positions here directly.

Reply:
When trying `const words = text.matchAll(anyWord) || [];` I get an Syntaxerror followed by this errormessage:

Property 'matchAll' does not exist on type 'string'

Reply:
Does `const words = anyWord.matchAll(text)` work?

Reply:
In that case I get the same error message, only now "string" is replaced by "RegExp"

Reply:
Reason is that `matchAll` is pretty new: https://stackoverflow.com/questions/55691536/does-node-js-support-the-string-matchall-method/55691566.
[Here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/matchAll#Browser_compatibility) it actually states that node.js (the engine running the tests) does not support it yet. 
We **could** add a polyfill to enable it, but I doubt it's worth the effort for now. I you want to spend a few minutes on that (it probably means installing / updating a few npm packages) feel free to do so, but if you bite your teeth out on it for more than 10min, leave it be I'd say.

Reply:
I found this: https://babeljs.io/docs/en/babel-polyfill. Is this here all I need to install?

Reply:
maybe. Maybe `core-js` would be fine as well. I'll leave that to you to experiment on in a separate branch :)